testing/testSortAlgorithm.py

```diff
@@ -0,0 +1,59 @@
+import os
+import sys
+sys.path.insert(0, '/path/to/sortAlgorithm')
+# print(os.getcwd())
+import sortingHat
+import unittest
+import tempfile
+
+class SortingHatTest(unittest.TestCase):
+
+    # def setUp(self):
+    #     #create new test client and initialize new db
+    #     self.db_fd, sortingHat.app.config['DATABASE'] = tempfile.mkstemp()
+    #     #diable error catching during request handling
+    #     sortingHat.app.config['TESTING'] = True
+    #     self.app = sortingHat.app.test_client()
+    #     with sortingHat.app.app_context():
+    #         sortingHat.init_db()
+
+    def test_CalcStudentPreference(self):
+        test1 = sortingHat.calcStudentPreference(.3, .3, .3)
+        self.assertEqual(test1, 1.6985348)
+        # test illegal arguments
+        with self.assertRaises(AssertionError):
+            sortingHat.calcStudentPreference(-1, 1, 1)
+
+    def test_CalcGroupPreference(self):
+        test1 = sortingHat.calcGroupPreference(.3, .3)
+        self.assertEqual(test1, 0.1)
+        # test illegal arguments
+        with self.assertRaises(AssertionError):
+            sortingHat.calcGroupPreference(-1, 1)
+
+    def test_GroupSizeWithin(self):
```

Comment:
I like the bones and the comments! I feel (as documentation lead) that these functions should have comments above documenting what they're testing specifically (but obviously that can be done on the final merge 😄 )